pkg/profiling/profiling.go

```diff
@@ -0,0 +1,94 @@
+package profiling
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"runtime"
+	"runtime/pprof"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/pkg/logging"
+)
+
+var (
+	log                   = logging.LoggerForModule()
+	heapdumpSubfolderName = "heapdump"
+	// time to wait between heap dumps when hitting threshold in seconds
+	defaultHeapProfilerBackoff = 30
+)
+
+// HeapProfiler is used to start a ticker that periodically checks if heap memory consumption
+// exceed the thresholdFraction, if so the heap gets dumped to a File in Directory.
+type HeapProfiler struct {
+	// Backoff limits the maximum frequency of creating heap dumps
+	Backoff           time.Duration
```

Comment:
I'm a little lost, why would we want to tweak the `Backoff` field on a live `HeapProfiler` object from a different package? :thinking:

Reply:
If I'd use a HeapProfiler from another package in my program, I would want this to be configurable from the caller site e.g. with env variables for the binary. Let's say we want to analyze a binary but it dies within 40 seconds because of OOM, I'd want to be able to set the backoff to 10 secs to get multiple dumps in this timeframe. IMO we have 3 ways to allow that:

1. Extend the NewHeapProfiler function with a backoff parameter
       - we can add it here, so that the whole config is done with the NewHeapProfiler and we don't mix different types of field access here
2. Make the field public 
       - We have a good default, so why should we extend the NewHeapProfiler function, this makes it more complicated, just to support a use case that is probably not used very often. That's why I took this option
3. Use Setter/Getter for this
       - I personally don't like getter/setter just for the purpose of making a field private

Reply:
I'd definitely go for (1), if only not to have to worry about thread-safety issues.